internal/handlers/common/aggregations/operators/operators.go

```diff
@@ -98,19 +86,41 @@ func NewOperator(doc any) (Operator, error) {
 
 	switch {
 	case !operatorExists:
-		return nil, ErrNoOperator
+		return nil, newOperatorError(
+			ErrNoOperator,
+			fmt.Errorf("No operator in document"),
+		)
+
 	case operatorDoc.Len() > 1:
-		return nil, ErrTooManyFields
+		return nil, newOperatorError(
+			ErrTooManyFields,
+			fmt.Errorf("The operator field specifies more than one operator"),
+		)
 	}
 
 	operator := operatorDoc.Command()
 
-	newOperator, ok := Operators[operator]
-	if !ok {
-		return nil, ErrNotImplemented
-	}
+	newOperator, supported := Operators[operator]
+	_, unsupported := unsupportedOperators[operator]
 
-	return newOperator(operatorDoc)
+	switch {
+	case supported && unsupported:
+		panic(fmt.Sprintf("operator %q is in both `operators` and `unsupportedOperators`", operator))
+	case supported && !unsupported:
+		return newOperator(operatorDoc)
+	case !supported && unsupported:
+		return nil, newOperatorError(
+			ErrNotImplemented,
+			fmt.Errorf("The operator %s is not implemented yet", operator),
+		)
+	case !supported && !unsupported:
+		return nil, newOperatorError(
+			ErrInvalidExpression,
+			fmt.Errorf("Unrecognized expression '%s'", operator),
+		)
+	default:
```

Comment:
Minor suggestion of simplifying. I think we don't need panic, if it's in `supported` let's return new operator. This sway it reduces untestable code by integration test. 
```
case supported:
     return newOperator(operatorDoc)
case !unsupported:
     ErrInvalidExpression
default:
     ErrNotImplemented
```

Reply:
You liked it but there is no change in the code? :thinking:

Reply:
There is a change in code, but maybe I misunderstood the comment.

The old code:
```switch {
	case supported && unsupported:
		panic(fmt.Sprintf("operator %q is in both `operators` and `unsupportedOperators`", operator))
	case supported && !unsupported:
		return newOperator(operatorDoc)
	case !supported && unsupported:
		return nil, newOperatorError(
			ErrNotImplemented,
			fmt.Errorf("The operator %s is not implemented yet", operator),
		)
	case !supported && !unsupported:
		return nil, newOperatorError(
			ErrInvalidExpression,
			fmt.Errorf("Unrecognized expression '%s'", operator),
		)
	default:
		panic("Unreachable")
	}
```

The new code:
```
	switch {
	case supported && unsupported:
		panic(fmt.Sprintf("operator %q is in both `operators` and `unsupportedOperators`", operator))
	case supported && !unsupported:
		return newOperator(operatorDoc)
	case !supported && unsupported:
		return nil, newOperatorError(
			ErrNotImplemented,
			fmt.Sprintf("The operator %s is not implemented yet", operator),
		)
	default:
		return nil, newOperatorError(
			ErrInvalidExpression,
			fmt.Sprintf("Unrecognized expression '%s'", operator),
		)
	}

```

I've replaced the unreachable panic and simplified the switch a bit.

Reply:
Oh, maybe you were talking about the panic about operator that is both supported and unsupported? We have similar logic in [stages](https://github.com/FerretDB/FerretDB/blob/main/internal/handlers/common/aggregations/stages/stages.go#L95-L115). I think it's worth to catch if operator is removed from `unsupportedOperators` list after introducing support for it. If we think otherwise then we can get rid off the second panic :)

Reply:
Ah yeah never mind, I didn't explain it well enough. Let's leave it and at some point we may or may not :hugs: I was just asking why not have supported case, unsupported case and default case for not implemented, ignoring `supported && unsupported`,